### PR TITLE
Tutorial - Build One Productivity Hub app by using Microsoft Graph Toolkit 

### PR DESCRIPTION
**Pull request by aycabas**
Added the following files to create the new tutorial:
  yml file:
  - concepts/graph-toolkit-building-one-productivity-hub.yml
  markdown files:
  - concepts/includes/graph-toolkit-building-one-productivity-hub-01-intro.md
  - concepts/includes/graph-toolkit-building-one-productivity-hub-02-register-app.md
  - concepts/includes/graph-toolkit-building-one-productivity-hub-03-build-app.md
  - concepts/includes/graph-toolkit-building-one-productivity-hub-04-design-app.md
  - concepts/includes/graph-toolkit-building-one-productivity-hub-05-run-app.md
  - concepts/includes/graph-toolkit-building-one-productivity-hub-06-summary.md
  images folder:
  - concepts/images/mgt-one-productivity-hub

### CODE REVIEW

File: concepts/includes/graph-toolkit-building-one-productivity-hub-03-build-app.md [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+1. Create a new file under **OneProductivityHub** folder and name as **index.html**.
+
+1. Select `CTRL + SPACE` and choose **HTML sample** from the options.
+
+## Enable authentication with the Microsoft Graph Toolkit
+
+To use the Microsoft Graph Toolkit via mgt-loader, add the following reference in **index.html** under the `<head></head>` section:
+
+```HTML
+<script src="https://unpkg.com/@microsoft/mgt@2.5.1/dist/bundle/mgt-loader.js"></script>
```
Comment (sebastienlevert) [inline concepts/includes/graph-toolkit-building-one-productivity-hub-03-build-app.md:16]: Please the latest 2.6.0 version!

File: concepts/includes/graph-toolkit-building-one-productivity-hub-06-summary.md [hunk 2]
```diff
@@ -0,0 +1,5 @@
+You have successfully completed the Build One Productivity Hub app by using Microsoft Graph Toolkit tutorial: you created a web app and consumed Microsoft Graph Toolkit MSAL2 provider and UI components.
+
+## Next steps
```
Comment (sebastienlevert) [inline concepts/includes/graph-toolkit-building-one-productivity-hub-06-summary.md:3]: Should we invite them to integrate more components that weren't included in the tutorial?

File: concepts/includes/graph-toolkit-building-one-productivity-hub-06-summary.md [hunk 3]
```diff
@@ -0,0 +1,5 @@
...
+
+To find out more about Microsoft Graph Toolkit, visit [Microsoft Graph Toolkit overview](https://docs.microsoft.com/en-us/graph/toolkit/overview).
```
Comment (sebastienlevert) [inline concepts/includes/graph-toolkit-building-one-productivity-hub-06-summary.md:5]: Maybe a link to the MS Learn module?

File: concepts/includes/graph-toolkit-building-one-productivity-hub-01-intro.md [hunk 4]
```diff
@@ -0,0 +1,15 @@
+Microsoft Graph Toolkit is a collection of web components and authentication providers for connecting apps to Microsoft 365 data and intelligence.
```
Comment (sebastienlevert) [inline concepts/includes/graph-toolkit-building-one-productivity-hub-01-intro.md:1]: Should we add a link to the "finished" tutorial repo?

**Review by sebastienlevert — CHANGES_REQUESTED** [review]

File: concepts/includes/graph-toolkit-building-one-productivity-hub-03-build-app.md (hunk 1 above)
Comment (aycabas) [reply concepts/includes/graph-toolkit-building-one-productivity-hub-03-build-app.md:16]: @sebastienlevert Login component doesn't initiate the authentication in version 2.6.0 and 2.5.2, tested in Chrome & Edge. UI is there but it's not clickable. 2.5.1 works stable with Login component and MSAL2 Provider. Should we proceed with this one right now, and update it after fixing the issue? Shall I report this bug for MGT?

File: concepts/includes/graph-toolkit-building-one-productivity-hub-06-summary.md (hunk 3 above)
Comment (aycabas) [reply concepts/includes/graph-toolkit-building-one-productivity-hub-06-summary.md:5]: Great idea, done!

File: concepts/includes/graph-toolkit-building-one-productivity-hub-01-intro.md (hunk 4 above)
Comment (aycabas) [reply concepts/includes/graph-toolkit-building-one-productivity-hub-01-intro.md:1]: This tutorial contains everthing available in the official OfficeDev repo right now except the project itself. Maybe I can remove the readme files from the repo and only share the solution files to avoid confusion?

File: concepts/includes/graph-toolkit-building-one-productivity-hub-03-build-app.md (hunk 1 above)
Comment (aycabas) [reply concepts/includes/graph-toolkit-building-one-productivity-hub-03-build-app.md]: updated the code as suggested and changed the version to 2.6.0.

File: concepts/graph-toolkit-building-one-productivity-hub.yml [hunk 5]
```diff
@@ -0,0 +1,38 @@
+### YamlMime:Tutorial
+title: Build One Productivity Hub app by using Microsoft Graph Toolkit
```
Comment (jasonjoh) [inline concepts/graph-toolkit-building-one-productivity-hub.yml:2]: "One Productivity Hub" is the name of the sample app you created, is that right? I'm wondering if we should change the title to something more universal, like "Build productivity apps by using...". I'm just concerned that "Build One Productivity Hub" at a glance just raises the question "What is One Productivity Hub?" :D

File: concepts/includes/graph-toolkit-building-one-productivity-hub-01-intro.md [hunk 6]
```diff
@@ -0,0 +1,15 @@
...
+
```
Comment (jasonjoh) [inline concepts/includes/graph-toolkit-building-one-productivity-hub-01-intro.md:2]: Combine these two sentences into one paragraph.

File: concepts/includes/graph-toolkit-building-one-productivity-hub-01-intro.md [hunk 7]
```diff
@@ -0,0 +1,15 @@
...
+This tutorial shows you how to create a web application to monitor your calendar events, to-do tasks and files by using Microsoft Graph Toolkit components and MSAL2 provider.
+
+## How does the sample work?
+
+This sample creates a web app that consumes Microsoft Graph Toolkit MSAL2 provider to enable authentication through Azure Active Directory and uses UI components to demonstrate calendar events, to-do tasks and files that will look and feel like Microsoft experiences out of the box.
```
Comment (jasonjoh) [inline concepts/includes/graph-toolkit-building-one-productivity-hub-01-intro.md:7]: "out of the box" may not translate well. I'd rephrase this to "...look and feel like a native Microsoft experience."

File: concepts/includes/graph-toolkit-building-one-productivity-hub-02-register-app.md [hunk 8]
```diff
@@ -0,0 +1,17 @@
+After all the prerequisites are in place, you will be able to register an application in the Azure Active Directory. The registration is necessary in order to authenticate the application through Microsoft Graph Toolkit MSAL2 provider and use UI components.
```
Comment (jasonjoh) [inline concepts/includes/graph-toolkit-building-one-productivity-hub-02-register-app.md:1]: I'd rephrase this a bit with a more active voice. You don't need all the pre-reqs to register an app :D.

"Register an application in the Azure Active Directory to enable user authentication for your application."

File: concepts/includes/graph-toolkit-building-one-productivity-hub-02-register-app.md [hunk 9]
```diff
@@ -0,0 +1,17 @@
...
+1. Select **App Registration** and click on **New Registration**.
+
+   ![Screenshot showing Azure Active Directory App registration tab](../images/mgt-one-productivity-hub/aad-app-registration.png)
+
+1. Complete the **Register an application** form with the following values, and then select **Register**.
+   * **Name**: One Productivity Hub Demo
+   * **Supported account types**: Accounts in any organizational directory (Any Azure AD directory - Multitenant) and personal Microsoft accounts (e.g. Skype, Xbox)
+   * **Redirect URI**: Select **Single-page application (SPA)** as a type of redirect URI and place `http://localhost:3000/index.html` as a redirect URI to test locally.
+
+    ![Screenshot showing the new app registration form](../images/mgt-one-productivity-hub/aad-new-app-registration-form.png)
```
Comment (jasonjoh) [inline concepts/includes/graph-toolkit-building-one-productivity-hub-02-register-app.md:15]: The Docs guidelines recommend when doing multiple screenshots from a website, you have the first one show a full browser window, but subsequent ones can be smaller snips. Not a crucial thing, but you may want to trim this second screenshot down a bit.

I'd also recommend removing the redlines on this screenshot - it's a pretty linear page flow so I don't think user needs that visual guidance.

File: concepts/includes/graph-toolkit-building-one-productivity-hub-03-build-app.md [hunk 10]
```diff
@@ -0,0 +1,74 @@
...
+
+1. Create a new folder and name as **OneProductivityHub**. Right click and open the folder with Visual Studio Code.  
+
+1. Create a new file under **OneProductivityHub** folder and name as **index.html**.
+
+1. Select `CTRL + SPACE` and choose **HTML sample** from the options.
+
+## Enable authentication with the Microsoft Graph Toolkit
+
+To use the Microsoft Graph Toolkit via mgt-loader, add the following reference in **index.html** under the `<body></body>` section:
```
Comment (jasonjoh) [inline concepts/includes/graph-toolkit-building-one-productivity-hub-03-build-app.md:13]: I would suggest removing the H3 below and adding this into the numbered list.

File: concepts/includes/graph-toolkit-building-one-productivity-hub-03-build-app.md [hunk 11]
```diff
@@ -0,0 +1,74 @@
...
+```HTML
+<script src="https://unpkg.com/@microsoft/mgt@2.6.0/dist/bundle/mgt-loader.js"></script>
+```
+
+### Initialize the MSAL2 provider
+
+1. In **index.html**, add the MSAL2 provider in the `<body></body>` section as following:
+
+    ```HTML
+    <mgt-msal2-provider
```
Comment (jasonjoh) [inline concepts/includes/graph-toolkit-building-one-productivity-hub-03-build-app.md:24]: Some of these scopes are redundant, though I don't know if maybe there's some MGT requirement of doing it this way. 

- Files.Read should be included in Files.Read.All (I believe)
- Tasks.Read should be included in Tasks.ReadWrite
- User.ReadBasic.All is listed twice

File: concepts/graph-toolkit-building-one-productivity-hub.yml [hunk 12]
```diff
@@ -0,0 +1,38 @@
...
+items:
+  - title: Build One Productivity Hub app by using Microsoft Graph Toolkit
+    durationInMinutes: 5
+    content: |
+      [!INCLUDE [](includes/graph-toolkit-building-one-productivity-hub-01-intro.md)]
+  - title: Register the application in Azure Active Directory
+    durationInMinutes: 10
+    content: |
+      [!INCLUDE [](includes/graph-toolkit-building-one-productivity-hub-02-register-app.md)]
+  - title: Build the application
```
Comment (jasonjoh) [inline concepts/graph-toolkit-building-one-productivity-hub.yml:23]: I'd change this to "Create the application"

File: concepts/includes/graph-toolkit-building-one-productivity-hub-04-design-app.md [hunk 13]
```diff
@@ -0,0 +1,190 @@
+In this step, you'll design your web app by using Microsoft Graph Toolkit components and style it with CSS.
+
+## Initialize the Login component
+
+In **index.html** inside the `<body></body>`, create `<div></div>` right under the provider and add the Login component inside the div:
```
Comment (jasonjoh) [inline concepts/includes/graph-toolkit-building-one-productivity-hub-04-design-app.md:5]: I'd add the <div> into your code block below.

File: concepts/includes/graph-toolkit-building-one-productivity-hub-04-design-app.md [hunk 14]
```diff
@@ -0,0 +1,190 @@
...
+
+  ```html
+  <mgt-login></mgt-login>
```
Comment (jasonjoh) [inline concepts/includes/graph-toolkit-building-one-productivity-hub-04-design-app.md:8]: Can this be simplified as `<mgt-login/>`?

File: concepts/includes/graph-toolkit-building-one-productivity-hub-05-run-app.md [hunk 15]
```diff
@@ -0,0 +1,15 @@
+In this step, you will build and run the sample. This code sample be initialized by Live Server and One Productivity Hub page will pop up in your browser.
```
Comment (jasonjoh) [inline concepts/includes/graph-toolkit-building-one-productivity-hub-05-run-app.md:1]: I think there's a typo or missing word in the second sentence.

File: concepts/includes/graph-toolkit-building-one-productivity-hub-05-run-app.md [hunk 16]
```diff
@@ -0,0 +1,15 @@
...
+
+1. Select `CTRL + SHIFT + P` to open the panel in Visual Studio Code, type and choose `Live Server: Open with Live Server`. Live Server will run the app in your browser.
+
+1. Select **Sign in** and use your Microsoft 365 Developer Program tenant to login.
+
+    ![Screenshot showing the One Productivity Hub initial page](../images/mgt-one-productivity-hub/run-app-sign-in.png)
+
+1. Consent the required permissions to use the application features such as viewing calendar events,to-do tasks and file folders.
```
Comment (jasonjoh) [inline concepts/includes/graph-toolkit-building-one-productivity-hub-05-run-app.md:9]: Three things on this one:

- "Consent to the required..."
- missing space after comma
- Missing that Oxford comma (per MS style guide)

File: concepts/includes/graph-toolkit-building-one-productivity-hub-05-run-app.md [hunk 17]
```diff
@@ -0,0 +1,15 @@
+In this step, you will build and run the sample. This code sample be initialized by Live Server and One Productivity Hub page will pop up in your browser.
+
+1. Select `CTRL + SHIFT + P` to open the panel in Visual Studio Code, type and choose `Live Server: Open with Live Server`. Live Server will run the app in your browser.
+
+1. Select **Sign in** and use your Microsoft 365 Developer Program tenant to login.
+
+    ![Screenshot showing the One Productivity Hub initial page](../images/mgt-one-productivity-hub/run-app-sign-in.png)
```
Comment (jasonjoh) [inline concepts/includes/graph-toolkit-building-one-productivity-hub-05-run-app.md:7]: Overall this page is very screenshot heavy with single sentences between. I would recommend cutting all but the last one.

**Review by jasonjoh — CHANGES_REQUESTED** [review]

**Comment by jasonjoh** [comment]
Thanks for doing this Ayca!

File: concepts/graph-toolkit-building-one-productivity-hub.yml (hunk 5 above)
Comment (aycabas) [reply concepts/graph-toolkit-building-one-productivity-hub.yml:2]: Makes more sense. Updated the title as "Build productivity apps by using Microsoft Graph Toolkit".

File: concepts/includes/graph-toolkit-building-one-productivity-hub-01-intro.md (hunk 6 above)
Comment (aycabas) [reply concepts/includes/graph-toolkit-building-one-productivity-hub-01-intro.md:2]: Done.

File: concepts/includes/graph-toolkit-building-one-productivity-hub-01-intro.md (hunk 7 above)
Comment (aycabas) [reply concepts/includes/graph-toolkit-building-one-productivity-hub-01-intro.md:7]: Done.

File: concepts/includes/graph-toolkit-building-one-productivity-hub-02-register-app.md (hunk 8 above)
Comment (aycabas) [reply concepts/includes/graph-toolkit-building-one-productivity-hub-02-register-app.md:1]: Done :)

File: concepts/includes/graph-toolkit-building-one-productivity-hub-02-register-app.md (hunk 9 above)
Comment (aycabas) [reply concepts/includes/graph-toolkit-building-one-productivity-hub-02-register-app.md:15]: Updated the screenshot.

File: concepts/includes/graph-toolkit-building-one-productivity-hub-03-build-app.md (hunk 10 above)
Comment (aycabas) [reply concepts/includes/graph-toolkit-building-one-productivity-hub-03-build-app.md:13]: Done.

File: concepts/includes/graph-toolkit-building-one-productivity-hub-03-build-app.md (hunk 11 above)
Comment (aycabas) [reply concepts/includes/graph-toolkit-building-one-productivity-hub-03-build-app.md:24]: Updated.

File: concepts/graph-toolkit-building-one-productivity-hub.yml (hunk 12 above)
Comment (aycabas) [reply concepts/graph-toolkit-building-one-productivity-hub.yml:23]: Done

File: concepts/includes/graph-toolkit-building-one-productivity-hub-04-design-app.md (hunk 13 above)
Comment (aycabas) [reply concepts/includes/graph-toolkit-building-one-productivity-hub-04-design-app.md:5]: done

File: concepts/includes/graph-toolkit-building-one-productivity-hub-04-design-app.md (hunk 14 above)
Comment (aycabas) [reply concepts/includes/graph-toolkit-building-one-productivity-hub-04-design-app.md:8]: Unfortunately, we need to use the component as `<mgt-login></mgt-login>`. Tested the suggested way, UI doesn't appear.

File: concepts/includes/graph-toolkit-building-one-productivity-hub-05-run-app.md (hunk 15 above)
Comment (aycabas) [reply concepts/includes/graph-toolkit-building-one-productivity-hub-05-run-app.md:1]: Updated the sentences.

File: concepts/includes/graph-toolkit-building-one-productivity-hub-05-run-app.md (hunk 16 above)
Comment (aycabas) [reply concepts/includes/graph-toolkit-building-one-productivity-hub-05-run-app.md:9]: Updated.

File: concepts/includes/graph-toolkit-building-one-productivity-hub-05-run-app.md (hunk 17 above)
Comment (aycabas) [reply concepts/includes/graph-toolkit-building-one-productivity-hub-05-run-app.md:7]: Removed images for *Sign in* and *Permissions* processes.

**Comment by aycabas** [comment]
Thanks for all the comments @jasonjoh, updated the tutorial accordingly. Please review the final version and let me know if you recommend any other changes. Thank you!

File: concepts/includes/graph-toolkit-building-one-productivity-hub-03-build-app.md [hunk 18]
```diff
@@ -0,0 +1,66 @@
...
+
+1. Create a new folder and name as **OneProductivityHub**. Right click and open the folder with Visual Studio Code.  
+
+1. Create a new file under **OneProductivityHub** folder and name as **index.html**.
+
+1. Select `CTRL + SPACE` and choose **HTML sample** from the options.
+
+1. Enable authentication with the Microsoft Graph Toolkit via mgt-loader, add the following reference in **index.html** under the `<body></body>` section:
+
+```HTML
```
Comment (jasonjoh) [inline concepts/includes/graph-toolkit-building-one-productivity-hub-03-build-app.md:13]: indent this block so it lines up with the last step

File: concepts/includes/graph-toolkit-building-one-productivity-hub-03-build-app.md [hunk 19]
```diff
@@ -0,0 +1,66 @@
...
+        Calendars.Read,
+        Files.Read.All,
+        Sites.Read.All,
+        Tasks.ReadWrite,
+        People.Read">    
+    </mgt-msal2-provider>
+    ```
+
+    > [!IMPORTANT]
+    > The following scopes defined in the provider will be shown as a list of required permissions to request user's consent during the authentication process: `User.Read, User.ReadBasic.All, Calendars.Read, Files.Read, Files.Read.All, Sites.Read.All, Tasks.Read, Tasks.ReadWrite, People.Read, User.ReadBasic.All`.
```
Comment (jasonjoh) [inline concepts/includes/graph-toolkit-building-one-productivity-hub-03-build-app.md:35]: Update the list of permissions here to match the code above.

File: concepts/includes/graph-toolkit-building-one-productivity-hub-03-build-app.md [hunk 20]
```diff
@@ -0,0 +1,66 @@
...
+
+## Create a web app
+
+1. Create a new folder and name as **OneProductivityHub**. Right click and open the folder with Visual Studio Code.  
+
+1. Create a new file under **OneProductivityHub** folder and name as **index.html**.
+
+1. Select `CTRL + SPACE` and choose **HTML sample** from the options.
+
+1. Enable authentication with the Microsoft Graph Toolkit via mgt-loader, add the following reference in **index.html** under the `<body></body>` section:
```
Comment (jasonjoh) [inline concepts/includes/graph-toolkit-building-one-productivity-hub-03-build-app.md:11]: I would change to "inside the `<body></body>`"

File: concepts/includes/graph-toolkit-building-one-productivity-hub-03-build-app.md [hunk 21]
```diff
@@ -0,0 +1,66 @@
...
+      <meta charset='utf-8'>
+      <meta http-equiv='X-UA-Compatible' content='IE=edge'>
+      <title>One Productivity Hub</title>
+      <meta name='viewport' content='width=device-width, initial-scale=1'>
+      <link rel='stylesheet' type='text/css' media='screen' href='main.css'>
+      <script src='main.js'></script>
+    </head>
+    <body>
+      <script src="https://unpkg.com/@microsoft/mgt@2.6.0/dist/bundle/mgt-loader.js"></script>
+     <mgt-msal2-provider
```
Comment (jasonjoh) [inline concepts/includes/graph-toolkit-building-one-productivity-hub-03-build-app.md:54]: can you fix the alignment? the `mgt-msal2-provider` element is off

File: concepts/includes/graph-toolkit-building-one-productivity-hub-04-design-app.md (hunk 13 above)
Comment (jasonjoh) [reply concepts/includes/graph-toolkit-building-one-productivity-hub-04-design-app.md]: Doh, just realized what I typed there didn't render :D. I meant to add the `<div>` bit into the code block.

File: concepts/includes/graph-toolkit-building-one-productivity-hub-04-design-app.md [hunk 22]
```diff
@@ -0,0 +1,187 @@
+In this step, you'll design your web app by using Microsoft Graph Toolkit components and style it with CSS.
+
+## Initialize the Login component
+
+In **index.html** inside the `<body></body>`, create `<div></div>` right under the provider and add the Login component inside the div into your code block below:
```
Comment (jasonjoh) [inline concepts/includes/graph-toolkit-building-one-productivity-hub-04-design-app.md:5]: My previous comment didn't render properly, so recommenting here. I was sugesting moving the `<div>` bit into the code block.

```suggestion
In **index.html** inside the `<body></body>`, add the following code under the provider.

````html
<div>
  <mgt-login></mgt-login>
</div>
````
```

File: concepts/includes/graph-toolkit-building-one-productivity-hub-04-design-app.md [hunk 23]
```diff
@@ -0,0 +1,187 @@
...
+
+Under div tagged with `class="row"`, add the File list component inside the third column div:
+
+  ```HTML
+  <mgt-file-list></mgt-file-list>
+  ```
+
+## Style your web app with CSS
+
+1. Create **index.css** file under your project and add below CCS code:
```
Comment (jasonjoh) [inline concepts/includes/graph-toolkit-building-one-productivity-hub-04-design-app.md:62]: "add the following CSS code" (typo in CCS)

File: concepts/includes/graph-toolkit-building-one-productivity-hub-04-design-app.md [hunk 24]
```diff
@@ -0,0 +1,187 @@
...
+      overflow: hidden;
+      padding: 5px;
+      margin-top: 5px;
+    }
+    #mgt-col:hover {
+      overflow-y: auto;
+    }
+    ```
+
+1. In **index.html** under `<head></head>`, define the stylesheet link `href` as **index.css**:
```
Comment (jasonjoh) [inline concepts/includes/graph-toolkit-building-one-productivity-hub-04-design-app.md:133]: "inside `<head></head>`"

File: concepts/includes/graph-toolkit-building-one-productivity-hub-04-design-app.md (hunk 14 above)
Comment (jasonjoh) [reply concepts/includes/graph-toolkit-building-one-productivity-hub-04-design-app.md:8]: I was able to get it to work for this and all of the MGT components. The key is there should be a space before the `/`, so `<mgt-login />`

File: concepts/includes/graph-toolkit-building-one-productivity-hub-02-register-app.md [hunk 25]
```diff
@@ -0,0 +1,17 @@
...
+1. Select **Azure Active Directory** from the left pane.
+
+1. Select **App Registration** and click on **New Registration**.
+
+   ![Screenshot showing Azure Active Directory App registration tab](../images/mgt-one-productivity-hub/aad-app-registration.png)
+
+1. Complete the **Register an application** form with the following values, and then select **Register**.
+   * **Name**: One Productivity Hub Demo
+   * **Supported account types**: Accounts in any organizational directory (Any Azure AD directory - Multitenant) and personal Microsoft accounts (e.g. Skype, Xbox)
+   * **Redirect URI**: Select **Single-page application (SPA)** as a type of redirect URI and place `http://localhost:3000/index.html` as a redirect URI to test locally.
```
Comment (jasonjoh) [inline concepts/includes/graph-toolkit-building-one-productivity-hub-02-register-app.md:13]: I got an authentication error based on wrong redirect URI. It seems that my Live Server install (which I just installed, so no changing of defaults) opens on port 5500, and it uses `127.0.0.1` instead of `localhost`. AAD requires it to match exactly (so it doesn't consider `127.0.0.1` and `localhost` to be the same).

This could get very complicated if we try to account for everyone's custom settings, but if we just assume the default, it won't work for some people. I suggest two options:

One way to do it would be to use the default value for the registration instructions (`http://127.0.0.1:5500/index.html`) and add a NOTE telling the dev if they've configured a different host and port they need to update the value accordingly.

Another way to do it would be to leave the redirect as it is and add an instruction in "creating the app" to add local settings, like this:

1. CTRL+SHIFT+P, select **Preferences: Open Workspace Settings (JSON)**.
2. In the **settings.json** file that opens, add the following code:

    ```json
    {
        "liveServer.settings.host": "localhost",
        "liveServer.settings.port": 3000
    }
    ```

Error message:

```
AADSTS50011: The redirect URI 'http://127.0.0.1:5500/index.html' specified in the request does not match the redirect URIs configured for the application '617f5aea-c443-4510-ae31-085242c3d568'. Make sure the redirect URI sent in the request matches one added to your application in the Azure portal. Navigate to https://aka.ms/redirectUriMismatchError to learn more about how to fix this.
```

File: concepts/includes/graph-toolkit-building-one-productivity-hub-03-build-app.md (hunk 19 above)
Comment (jasonjoh) [reply concepts/includes/graph-toolkit-building-one-productivity-hub-03-build-app.md:35]: Hmm maybe we were too hasty here. I just ran it this way and I ended up getting three consent prompts. It seems the task list explicitly wants `Tasks.Read` and the file list explicitly wants `Files.Read`.

**Review by jasonjoh — CHANGES_REQUESTED** [review]
Some fit-and-finish stuff but I also found a pretty major issue with Live Server and the auth. I added a few suggestions on how to handle it.

File: concepts/includes/graph-toolkit-building-one-productivity-hub-03-build-app.md (hunk 18 above)
Comment (aycabas) [reply concepts/includes/graph-toolkit-building-one-productivity-hub-03-build-app.md:13]: done

File: concepts/includes/graph-toolkit-building-one-productivity-hub-03-build-app.md (hunk 19 above)
Comment (aycabas) [reply concepts/includes/graph-toolkit-building-one-productivity-hub-03-build-app.md:35]: I am adding `Tasks.Read` and `Files.Read` back in the list.

Comment (aycabas) [reply concepts/includes/graph-toolkit-building-one-productivity-hub-03-build-app.md:35]: Also updated the **Important** part: removed duplicate `User.ReadBasic.All`.

File: concepts/includes/graph-toolkit-building-one-productivity-hub-03-build-app.md (hunk 20 above)
Comment (aycabas) [reply concepts/includes/graph-toolkit-building-one-productivity-hub-03-build-app.md:11]: done

File: concepts/includes/graph-toolkit-building-one-productivity-hub-03-build-app.md (hunk 21 above)
Comment (aycabas) [reply concepts/includes/graph-toolkit-building-one-productivity-hub-03-build-app.md:54]: done

File: concepts/includes/graph-toolkit-building-one-productivity-hub-04-design-app.md (hunk 13 above)
Comment (aycabas) [reply concepts/includes/graph-toolkit-building-one-productivity-hub-04-design-app.md]: commited your suggestion there. Apologies I didn't understand in the first place :)

File: concepts/includes/graph-toolkit-building-one-productivity-hub-04-design-app.md (hunk 23 above)
Comment (aycabas) [reply concepts/includes/graph-toolkit-building-one-productivity-hub-04-design-app.md:67]: Done

File: concepts/includes/graph-toolkit-building-one-productivity-hub-04-design-app.md (hunk 24 above)
Comment (aycabas) [reply concepts/includes/graph-toolkit-building-one-productivity-hub-04-design-app.md:138]: done

File: concepts/includes/graph-toolkit-building-one-productivity-hub-02-register-app.md (hunk 25 above)
Comment (aycabas) [reply concepts/includes/graph-toolkit-building-one-productivity-hub-02-register-app.md:13]: Great point, I added the second suggestion inside the run-app.md, so devs can setup settings.json before they run the app.

**Review by jasonjoh — APPROVED** [review]

**Comment by Lauragra** [comment]
> @Lauragra We completed Jason's review for this tutorial. Is there anything else needed for publishing?

@aycabas , @jasonjoh , this PR is still set to content review. If we're ready to publish, please se the ready to merge label. Or do we need an editorial review of the content? I can take a look. Thanks.

**Comment by aycabas** [comment]
> > @Lauragra We completed Jason's review for this tutorial. Is there anything else needed for publishing?
> 
> @aycabas , @jasonjoh , this PR is still set to content review. If we're ready to publish, please se the ready to merge label. Or do we need an editorial review of the content? I can take a look. Thanks.

That would be great to get your review @Lauragra, let me know if there are any more changes needed and I can update the tutorial.

**Review by isvargasmsft — APPROVED** [review]

File: concepts/includes/graph-toolkit-building-one-productivity-hub-01-intro.md [hunk 26]
```diff
@@ -0,0 +1,13 @@
+Microsoft Graph Toolkit is a collection of web components and authentication providers for connecting apps to Microsoft 365 data and intelligence. This tutorial shows you how to create a web application to monitor your calendar events, to-do tasks and files by using Microsoft Graph Toolkit components and MSAL2 provider.
+
+## How does the sample work?
+
+This sample creates a web app that consumes Microsoft Graph Toolkit MSAL2 provider to enable authentication through Azure Active Directory and uses UI components to demonstrate calendar events, to-do tasks and files that will look and feel like native Microsoft experiences.
```
Comment (Lauragra) [inline concepts/includes/graph-toolkit-building-one-productivity-hub-01-intro.md]: What do we mean by "demonstrate" calendar events and so on? Does the web app "render" those items? Is it the sample or the web app that is the subject of the verb "demonstrate"? Please help to clarify this sentence.

File: concepts/includes/graph-toolkit-building-one-productivity-hub-05-run-app.md [hunk 27]
```diff
@@ -0,0 +1,22 @@
+In this step, you will run your sample app in the browser by using Live Server.
+
+1. `CTRL + SHIFT + P`, type and select `Preferences: Open Workspace Settings (JSON)`.
```
Comment (Lauragra) [inline concepts/includes/graph-toolkit-building-one-productivity-hub-05-run-app.md]: Are we missing something that they need to type, or do they just select something? The phrase "type and select" is a little unclear.

File: concepts/includes/graph-toolkit-building-one-productivity-hub-05-run-app.md [hunk 28]
```diff
@@ -0,0 +1,22 @@
...
+1. In the `settings.json` file that opens, add the following code:
+
+    ```json
+    {
+    "liveServer.settings.host": "localhost",
+    "liveServer.settings.port": 3000
+    }
+    ```
+
+1. Select `CTRL + SHIFT + P` to open the panel in Visual Studio Code, type and choose `Live Server: Open with Live Server`. Live Server will run the app in your browser.
```
Comment (Lauragra) [inline concepts/includes/graph-toolkit-building-one-productivity-hub-05-run-app.md:14]: Here we say "type and choose", it's not clear if they type something or just choose something.

File: concepts/includes/graph-toolkit-building-one-productivity-hub-06-summary.md [hunk 29]
```diff
@@ -0,0 +1,7 @@
+You have successfully completed the Build One Productivity Hub app by using Microsoft Graph Toolkit tutorial: you created a web app and consumed Microsoft Graph Toolkit MSAL2 provider and UI components.
```
Comment (Lauragra) [inline concepts/includes/graph-toolkit-building-one-productivity-hub-06-summary.md:1]: Let's revise this sentence to be more clear. Suggested revision:

You have successfully built your One Productivity Hub sample app using Microsoft Graph Toolkit. In this tutorial, you created a web app and consumed the Microsoft Graph Toolkit MSAL2 provider and UI components.

**Review by Lauragra — COMMENTED** [review]
Thanks for creating this tutorial; looks good! Please see my review comments and clarify as appropriate. Thanks!

File: concepts/includes/graph-toolkit-building-one-productivity-hub-01-intro.md (hunk 26 above)
Comment (aycabas) [reply concepts/includes/graph-toolkit-building-one-productivity-hub-01-intro.md]: I updated the word "demonstrate" with "render". The sample renders calendar events, todo tasks and files.

File: concepts/includes/graph-toolkit-building-one-productivity-hub-05-run-app.md (hunk 27 above)
Comment (aycabas) [reply concepts/includes/graph-toolkit-building-one-productivity-hub-05-run-app.md]: updated the sentences as "Select `CTRL + SHIFT + P` to open the panel in Visual Studio Code, type and select `Preferences: Open Workspace Settings (JSON)`.
".  

We need to type on the panel to find the preferences and then select.

File: concepts/includes/graph-toolkit-building-one-productivity-hub-05-run-app.md (hunk 28 above)
Comment (aycabas) [reply concepts/includes/graph-toolkit-building-one-productivity-hub-05-run-app.md:14]: Updated this part as >> Select `CTRL + SHIFT + P` to open the panel in Visual Studio Code, type `Live Server` on the panel and select `Live Server: Open with Live Server`.

File: concepts/includes/graph-toolkit-building-one-productivity-hub-06-summary.md (hunk 29 above)
Comment (aycabas) [reply concepts/includes/graph-toolkit-building-one-productivity-hub-06-summary.md:1]: Updated as per your suggestion :)

**Comment by aycabas** [comment]
#feedback-addressed

**Review by Lauragra — APPROVED** [review]
